library/waku_thread/inter_thread_communication/requests/node_lifecycle_request.nim

```diff
@@ -43,149 +48,31 @@ proc destroyShared(self: ptr NodeLifecycleRequest) =
-  if mountArcRes.isErr():
-    return err("failed to mount waku archive protocol: " & mountArcRes.error)
-
-  # Store setup
-  try:
-    await mountStore(node)
-  except CatchableError:
-    return err("failed to mount waku store protocol: " & getCurrentExceptionMsg())
-
-  mountStoreClient(node)
-  if storeNode != "":
-    let storeNodeInfo = parsePeerInfo(storeNode)
-    if storeNodeInfo.isOk():
-      node.peerManager.addServicePeer(storeNodeInfo.value, WakuStoreCodec)
-    else:
-      return err("failed to set node waku store peer: " & storeNodeInfo.error)
-
-  return ok()
-
 proc createNode(configJson: cstring): Future[Result[WakuNode, string]] {.async.} =
-  var privateKey: PrivateKey
-  var netConfig = NetConfig.init(parseIpAddress("127.0.0.1"), Port(60000'u16)).value
-  ## relay
-  var relay: bool
-  var topics = @[""]
-
-  ## store
-  var store: bool
-  var storeNode: string
-  var storeRetentionPolicy: string
-  var storeDbUrl: string
-  var storeVacuum: bool
-  var storeDbMigration: bool
-  var storeMaxNumDbConnections: int
-
+  var conf: WakuNodeConf
   var errorResp: string
 
   try:
     if not parseConfig(
       $configJson,
-      privateKey,
-      netConfig,
-      relay,
-      topics,
-      store,
-      storeNode,
-      storeRetentionPolicy,
-      storeDbUrl,
-      storeVacuum,
-      storeDbMigration,
-      storeMaxNumDbConnections,
+      conf,
       errorResp,
     ):
       return err(errorResp)
   except Exception:
     return err("exception calling parseConfig: " & getCurrentExceptionMsg())
 
-  var enrBuilder = EnrBuilder.init(privateKey)
-
-  enrBuilder.withIpAddressAndPorts(
-    netConfig.enrIp, netConfig.enrPort, netConfig.discv5UdpPort
-  )
-
-  if netConfig.wakuFlags.isSome():
-    enrBuilder.withWakuCapabilities(netConfig.wakuFlags.get())
-
-  enrBuilder.withMultiaddrs(netConfig.enrMultiaddrs)
-
-  let addShardedTopics = enrBuilder.withShardedTopics(topics)
-  if addShardedTopics.isErr():
-    let msg = "Error setting shared topics: " & $addShardedTopics.error
-    return err(msg)
-
-  let recordRes = enrBuilder.build()
-  let record =
-    if recordRes.isErr():
-      let msg = "Error building enr record: " & $recordRes.error
-      return err(msg)
-    else:
-      recordRes.get()
-
-  ## TODO: make the next const configurable from 'configJson'.
-  const MAX_CONNECTIONS = 50.int
-
-  var builder = WakuNodeBuilder.init()
-  builder.withRng(crypto.newRng())
-  builder.withNodeKey(privateKey)
-  builder.withRecord(record)
-  builder.withNetworkConfiguration(netConfig)
-  builder.withSwitchConfiguration(maxConnections = some(MAX_CONNECTIONS))
-
-  let wakuNodeRes = builder.build()
-  if wakuNodeRes.isErr():
-    let errorMsg = "failed to create waku node instance: " & wakuNodeRes.error
-    return err(errorMsg)
-
-  var newNode = wakuNodeRes.get()
-
-  if relay:
-    await newNode.mountRelay()
-    newNode.peerManager.start()
+  # TODO: figure out how to extract default values from the config pragma
```

Comment:
I don't like `TODO`s in code. Either do it or open an issue. I feel it's hard to track.